webserver/pkgpkr/webservice/tests/test_recommender_service.py

```diff
@@ -0,0 +1,30 @@
+from unittest import TestCase
+import json
+from webservice.github_util import depenencies_name_to_purl
+from webservice.recommender_service import RecommenderService
+from .samples.sample_package_json import sample_package_json
+
+
+class TestRecommenderService(TestCase):
+
+    def setUp(self):
+        recommender_service = RecommenderService()
+
+        purl_dependencies = depenencies_name_to_purl(json.loads(sample_package_json)['dependencies'])
+
+        self.recommendations = recommender_service.get_recommendations(purl_dependencies)
+
+    def test_get_recommendations_count(self):
+        self.assertGreater(len(self.recommendations), 10)
+
+    def test_get_recommendation_content(self):
+        for recommendation in self.recommendations:
+            self.assertCountEqual({'name', 'url', 'average_downloads', 'keywords', 'date', 'rate'},
+                                  recommendation.keys())
+
+            self.assertIsNotNone(recommendation['name'])
+            self.assertIsNotNone(recommendation['url'])
+            # TODO this not always passing due to known issue in DB
+            # self.assertIsNotNone(recommendation['average_downloads'])
+            # self.assertIsNotNone(recommendation['date'])
```

Comment:
These should be passing now that the null metadata issue has been fixed, right?

Reply:
@djsauble 
`average_downloads` -  passing
`date` - no always.

Here's an example:
```
<class 'dict'>: 
{'name': 'pkg:npm/@croct-tech/sdk@0',
 'url': 'https://npmjs.com/package/@croct-tech/sdk', 
 'average_downloads': 0,
 'keywords': None,
 'date': None, 
 'rate': 0.0209564871153977}
```